mitosheet/mitosheet/api/get_code_snippets.py

```diff
@@ -62,5 +70,117 @@
+                        return f"Custom code snippets should have the format: {correct_code_snippets_type} {NEWLINE}{NEWLINE}But this code snippet was included: {code_snippet}"
+
+
+                id = code_snippet.get('Id', None)
+                name = code_snippet.get('Name', None)
+                description = code_snippet.get('Description', None)
+                code = code_snippet.get('Code', None)
+
+                error = False
+                if id is None or type(id) != str:
+                        error = True
+                if name is None or type(name) != str:
+                        error = True
+                if description is None or type(name) != str:
+                        error = True
+                if code is None or type(code) != list:
+                        error = True
+                if not error:
+                        for line in code:
+                                if type(line) != str:
+                                        error = True
+                                        break
+
+                if error:
+                        return f"Custom code snippets should have the format: {correct_code_snippets_type} {NEWLINE}{NEWLINE}But this code snippet was included: {code_snippet}"
+
+        return ''
+
+
+# Global variable used to cache the custom code snippets so that when users
+# open the code snippets taskpane multiple times in the same mito instantiation, 
+# we can display the custom code snippets without querying the url each time.
+CACHED_CUSTOM_CODE_SNIPPETS: Optional[List[CodeSnippet]] = None
+def get_custom_code_snippets(mito_config_code_snippets_url: str) -> str:
+        global CACHED_CUSTOM_CODE_SNIPPETS
+        if CACHED_CUSTOM_CODE_SNIPPETS is not None:
+                return json.dumps({
+                        'status': 'success',
+                        'code_snippets': CACHED_CUSTOM_CODE_SNIPPETS
+                })
+        
+        try:
+                # Request the code snippets from the url
+                response = req.get(mito_config_code_snippets_url, verify=False)
+        except Exception as e: 
+                return json.dumps({
+                        'status': 'error',
+                        'error_message': f"Error accessing the code snippets data from the URL. {e}" , 
+                })
+
+        if response.status_code == 200:
+                # Parse the respone body into JSON 
+                code_snippets = response.json()
+                code_snippet_format_error = get_code_snippets_format_error(code_snippets)
+
+                if code_snippet_format_error == '':
+                        # Cache the code snippets so we don't need to request them from the url next time
+                        CACHED_CUSTOM_CODE_SNIPPETS = code_snippets
+                        return json.dumps({
+                                'status': 'success',
+                                'code_snippets': code_snippets
+                        })
```

Comment:
Make this an object too so that we can make sure to update them all properly at once